### PR TITLE
tea: init at 0.7.0

### PR DESCRIPTION
**Pull request by j4m3s-s**
###### Motivation for this change
Add gitea cli tool

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/tea/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchgit {
+    url = "https://gitea.com/gitea/tea";
+    rev = version;
+    sha256 = "sha256-Kq+A6YELfBJ04t7pPnX8Ulh4NSMFn3AHggplLD9J8MY=";
+  };
+
+  vendorSha256 = null;
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tea/default.nix:15]: Please add a comment why you disabled tests.

File: pkgs/tools/misc/tea/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description = "Gitea official CLI client";
+    homepage    = "https://gitea.com/gitea/tea";
+    license     = licenses.mit;
+    platforms   = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tea/default.nix:21]:
```suggestion
```

already set by buildGoModule

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/tea/default.nix (hunk 1 above)
Comment (j4m3s-s) [reply pkgs/tools/misc/tea/default.nix]: Apparently I didn't need to disable them. I removed it.

File: pkgs/tools/misc/tea/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "tea";
+  version = "v0.7.0";
```
Comment (fabaff) [inline pkgs/tools/misc/tea/default.nix:5]: The numerical representation is enough.

```suggestion
  version = "0.7.0";
```

File: pkgs/tools/misc/tea/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchgit {
+    url = "https://gitea.com/gitea/tea";
+    rev = version;
```
Comment (fabaff) [inline pkgs/tools/misc/tea/default.nix:9]:
```suggestion
    rev = "v${version}";
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 116239` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tea</li>
  </ul>
</details>

**Pull request merged**